airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -545,3 +546,54 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
         session=session,
     )
     return task_instance_reference_collection_schema.dump(TaskInstanceReferenceCollection(task_instances=tis))
+
+
+@security.requires_access(
+    [
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG_RUN),
+    ],
+)
+@provide_session
+def set_task_instance_note(
```

Comment:
```suggestion
def set_task_instance_notes(
```

I think the name should match the field we set.

Reply:
Changed :)